src/lib/common/sol-types.c

```diff
@@ -174,11 +174,12 @@ sol_drange_equal(const struct sol_drange *var0, const struct sol_drange *var1)
     ((var1 > 0) && (var0 < INT32_MIN + var1))
 
 #define SOL_IRANGE_MULTIPLICATION_OVERFLOW(var0, var1) \
+    ((var0 == INT32_MAX) && (var1 == -1)) || \
     ((var0 > 0) && (var1 > 0) && (var0 > INT32_MAX / var1)) || \
     ((var0 < 0) && (var1 < 0) && (var0 < INT32_MAX / var1))
 
 #define SOL_IRANGE_MULTIPLICATION_UNDERFLOW(var0, var1) \
-    ((var0 > 0) && (var1 < 0) && (var0 > INT32_MIN / var1)) || \
+    ((var0 > 0) && (var1 < -1) && (var0 > INT32_MIN / var1)) || \
```

Comment:
If var1 == 0, the second condition fails, and since they are all &&, the division won't take place. All praise lazy evaluation.